src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java

```diff
@@ -45,6 +48,7 @@ public Oauth2ServiceImpl(
     this.appClientScopeRepository = appClientScopeRepository;
     this.appAccessTokenRepository = appAccessTokenRepository;
     this.appRefreshTokenRepository = appRefreshTokenRepository;
+    this.passwordEncoder = PasswordEncoderManager.createPasswordEncoder(passwordEncoderType);
```

Comment:
You don't need to do this every time you want to use the encoder. Same as in previous python implementation, add the `PasswordEncoder` instance to the DI container.

Reply:
Done.